apps/wakunode2/wakunode2.nim

```diff
@@ -87,26 +87,34 @@ when isMainModule:
   of inspectRlnDb:
     doInspectRlnDb(conf)
   of noCommand:
-    # The Waku Network config (cluster-id=1)
-    if conf.clusterId == 1:
-      let twnClusterConf = ClusterConf.TheWakuNetworkConf()
-      if len(conf.shards) != 0:
-        conf.pubsubTopics = conf.shards.mapIt(twnClusterConf.pubsubTopics[it.uint16])
+    case conf.clusterId
+      # cluster-id=0
+      of 0:
+        let clusterZeroConf = ClusterConf.ClusterZeroConf()
+        conf.pubsubTopics = clusterZeroConf.pubsubTopics
+        # TODO: Write some template to "merge" the configs
```

Comment:
What do you mean by "merge the configs" ?

Reply:
imagine
```
type A:
  field1
  field2

typeB:
  field1
  field2
  field3
  field4
```

which is similar to `WakuConf` and `ClusterZeroConf`. Note that `typeB` is a superset of `typeA`.

Right now we are "merging" boh manually.

```
let a = TypeA(...)
let b = TypeB(...)

# we want a config to override b
b.field1 = a.field1
b.field2 = a.field2
```

this is i) error prone, ii) if number of parameters increase, we have to assign them manually.

so I'm suggesting with some templating, writing a `merge()` function so that.
```
let a = TypeA(...)
let b = TypeB(...)

merge(a, b)
```

or something like this. with this we no longer need [all this](https://github.com/waku-org/nwaku/pull/2562/files#diff-fdd56fbeef6ba31430b71ea50a53439fc828b61e1d4291ad18e03906b41c7100R104-R115)